scripts/babel/transform-esm-compatability.js

```diff
@@ -0,0 +1,38 @@
+const checkForTypeCheck = (node) => {
```

Comment:
Small typo in the filename here: `compatability` => `compatibility`
Otherwise only the changeset is missing, but I suppose we can just `patch` bump everything

Reply:
I've been trying to test some stuff and it seems to work but we haven't got sandboxes for everything 😅 I think this code can't actually negatively impact any places judging from the astexplorer though. If anyone comes up with an edge case please throw it at me.

I'll add the changesets later